_plugins/rosindex_generator.rb

```diff
@@ -711,7 +711,7 @@ def scrape_repo(site, repo)
         unless explicit_version.nil?
           dputs (" Looking for explicit version #{explicit_version}").green
         end
-        version, snapshot.version = vcs.get_version(distro, explicit_version)
+        version, snapshot.version = vcs.get_version(explicit_version)
```

Comment:
The second output from this command looks to be set to be the input in L701 above. And we're returning `nil, nil` or `branch.name, branch_name` where we've confirmed that `branch_name == explict_version` ala L297 below. 

Thus I think we can reduce this to just a single return of the version instead and test that for nil anywhere necessary as snapshot.version is always explict_version unless nil when version is also nil

Reply:
Yes I can see that the to returned values are now always equal, so this code can be further reduced. I'll do that.

Reply:
Looking at this in more detail, I'm going to reverse what I said, and recommend against a change from my original patch here. snapshot.version is not always explicit version, for example in the Hg case, snapshot.version is 'default' when explicit_version is 'REMOTE_HEAD'. In the git version, it is more complex but not equal. Also, the two items returned are only both equal and strings in the Hg version. In the git version, the second item is a string, while the first is a Rugged branch object. It is possible that we could reduce all that to a single output, but that is really a refactor, not just eliminating redundant code.  Also, the snapshot.version that is returned from get_version may differ from the one earlier, as the name for example may be shortened by a split.